wazo_dird/plugins/source_result.py

```diff
@@ -104,25 +110,24 @@ def __repr__(self) -> str:
 
 
 def make_result_class(
-    source_backend,
-    source_name,
-    unique_column=None,
-    format_columns=None,
-    is_deletable=False,
-    is_personal=False,
-):
-    if not unique_column:
-        unique_column = _SourceResult._unique_column
-    if not format_columns:
-        format_columns = _SourceResult._format_columns
+    source_backend: str,
+    source_name: str,
+    unique_column: str | None = None,
+    format_columns: dict[str, str] | None = None,
+    is_deletable: bool = False,
+    is_personal: bool = False,
+) -> type[_SourceResult]:
+    unique_column = unique_column or _SourceResult._unique_column
+    format_columns = format_columns or _SourceResult._format_columns
+    _is_deletable = is_deletable
+    _is_personal = is_personal
 
     class SourceResult(_SourceResult):
         source = source_name
         backend = source_backend
         _unique_column = unique_column
         _format_columns = format_columns
-
-    SourceResult.is_deletable = is_deletable
-    SourceResult.is_personal = is_personal
+        is_deletable = _is_deletable
+        is_personal = _is_personal
```

Comment:
Why is the benefit to use intermediate variable `_is_delettable` instead of only use `is_deletable` (same as `source_name`)